scripts/check_requirements.py

```diff
@@ -11,15 +11,48 @@ def main():
             line.strip().split("#")[0].strip() for line in f.readlines()
         ]
 
-    installed_packages = [package.key for package in pkg_resources.working_set]
+    installed_packages = dict(
+        [(package.key, package.version) for package in pkg_resources.working_set]
+    )
+
+    pattern = (
+        r"^(?P<name>[^\s<>=]+)(?:\s*[@<>=]+\s*(?P<min_version>[^\s<>=]+))?(?:\s*,\s*<(?P<max_version>["
+        r"^\s<>=]+))?$"
+    )
+
+    url_pattern = r"(https?://[^\s/$.?#].[^\s]*)"
 
     missing_packages = []
     for package in required_packages:
         if not package:  # Skip empty lines
             continue
-        package_name = re.split("[<>=@ ]+", package.strip())[0]
-        if package_name.lower() not in installed_packages:
-            missing_packages.append(package_name)
+        package_data = re.match(pattern, package.strip())
```

Comment:
You can do this without all the `re` parsing with tools `pkg_resources` provides you:

```python

installed_packages = {pkg.key: pkg.version for pkg in pkg_resources.working_set}
missing_packages = []
for required_package in required_packages:
    pkg = pkg_resources.Requirement.parse(required_package)
    if (pkg.key not in installed_packages or
        pkg_resources.parse_version(installed_packages[pkg.key]) not in pkg.specifier):
        missing_packages.append(str(pkg))
```